src/wutil.cpp

```diff
@@ -564,7 +564,7 @@ file_id_t file_id_t::file_id_from_stat(const struct stat *buf)
 #if defined(__APPLE__) || defined(__DragonFly__) || defined(__FreeBSD__) ||  defined(__OpenBSD__) || defined(__NetBSD__)
     result.generation = buf->st_gen;
 #else
-    result.generation = 0;
+    result.generation = NULL;
```

Comment:
`NULL` should only be used with respect to pointers. The whole point of `NULL` was that, at least in the past, there were computer architectures that required a non-zero value to represent a pointer that could not be dereferenced. You'll find a lot of B.S. regarding this if you google questions like "c++ null pointer value". That's because people who haven't been programming for 30+ years and are therefore unlikely to have encountered unusual architectures like the 24 bit word length computers sold by Harris Computer Systems in the 1980's are unaware of the relevant history.